### ISSUE TITLE
HTML Accesskey doesn't work on MAC OS X

### ISSUE THREAD

**Issue by khushboovashi** [annulen/webkit#594]
Hi,

We have used your webkit in our project. I am facing the issue related to HTML access-keys.
On MAC OS X, the access keys don't work; I tried to dig into it and found out that the events are not getting generated.  For, example if I have used access-keys f to open the file menu then while I press (Alt + F), it should generate the mouse event which is not happening. 

Can you please help me to figure this out.

Thanks,
Khushboo

**Comment by annulen** [comment]
Did it work with older QtWebKit versions, e.g. one coming with Qt 5.5?

**Comment by khushboovashi** [comment]
MAC OS X - Doesn't work with official webkit 5.5 or newer version, not with Annulen 5.8 and also with webengine it doesn't work.
Linux - Works with everything
Windows: Works with Annulen 5.9 webkit

So, we have only issue on MAC OS X.

**Comment by annulen** [comment]
Is there any old report on this topic in bugreports.qt.io? I need to get as much details as possible, as I still don't realise what's all this about

**Comment by neel5481** [comment]
Issue was reported in 2009 as per below link but don't know it is integrated in Webkit source or not. 
https://bugs.webkit.org/show_bug.cgi?id=32213

**Comment by annulen** [comment]
Does iCtrl-Alt-F work?

**Comment by khushboovashi** [comment]
No. We have tried different key variations but didn't work.

**Comment by annulen** [comment]
I've tried test case https://github.com/annulen/webkit/blob/qtwebkit-stable/LayoutTests/fast/forms/access-key.html on Linux, works fine here.

Please try Command+Alt combination

**Comment by khushboovashi** [comment]
The problem is on the MAC OS X. On Linux it is working as expected.

**Comment by annulen** [comment]
Doesn't Command+Alt work?

If you can rebuild from sources I can give you a patch to try

**Comment by khushboovashi** [comment]
On Mon, Jul 31, 2017 at 7:53 PM, Konstantin Tokarev <
notifications@github.com> wrote:

> Doesn't Command+Alt work?
>
No. It doesn't work on MAC.

> If you can rebuild from sources I can give you a patch to try
>
Yes. we can do this. Can you please provide the steps/documentation link to
do this ?

I would like to clarify that, if this issue would be fixed with this patch,
then you will commit this patch to the repository.

**Comment by annulen** [comment]
> Can you please provide the steps/documentation link to do this ?

https://github.com/annulen/webkit/wiki/Building-QtWebKit-on-OS-X

>I would like to clarify that, if this issue would be fixed with this patch, then you will commit this patch to the repository.

Sure

Here is the patch:

```diff
diff --git a/Source/WebCore/page/qt/EventHandlerQt.cpp b/Source/WebCore/page/qt/EventHandlerQt.cpp
index 240ea75eb74..334ebbf6a3a 100644
--- a/Source/WebCore/page/qt/EventHandlerQt.cpp
+++ b/Source/WebCore/page/qt/EventHandlerQt.cpp
@@ -123,7 +123,7 @@ bool EventHandler::passMouseReleaseEventToSubframe(MouseEventWithHitTestResults&
 unsigned EventHandler::accessKeyModifiers()
 {
 #if OS(DARWIN)
-    return PlatformEvent::CtrlKey | PlatformEvent::AltKey;
+    return PlatformEvent::AltKey;
 #else
     return PlatformEvent::AltKey;
 #endif
```

**Comment by annulen** [comment]
With patch Alt+F is expected to be working

**Comment by khushboovashi** [comment]
On Tue, Aug 1, 2017 at 2:43 PM, Konstantin Tokarev <notifications@github.com
> wrote:

> With patch Alt+F is expected to be working
>
Thanks. I am currently trying to build webkit. I will let you know the
status once it's done.

**Comment by khushboovashi** [comment]
Hi,

The build was successful but when I run my application, I get below error.

dyld: Library not loaded:
/Users/username/Projects/qtwebkit/WebKitBuild/Release/lib/@rpath/QtWebKit.framework/Versions/5/QtWebKit

  Referenced from:
/Users/username/Qt5.9.1/5.9.1/clang_64/lib/QtWebKitWidgets.framework/Versions/5/QtWebKitWidgets

  Reason: image not found

Trace/BPT trap: 5

**Comment by khushboovashi** [comment]
On Tue, Aug 1, 2017 at 3:55 PM, khushboo Vashi <khushboo.vashi@gmail.com>
wrote:

> Hi,
>
> The build was successful but when I run my application, I get below error.
>
> dyld: Library not loaded: /Users/username/Projects/
> qtwebkit/WebKitBuild/Release/lib/@rpath/QtWebKit.framework/
> Versions/5/QtWebKit
>
>   Referenced from: /Users/username/Qt5.9.1/5.9.1/
> clang_64/lib/QtWebKitWidgets.framework/Versions/5/QtWebKitWidgets
>
>   Reason: image not found
>
> Trace/BPT trap: 5
>
>
I have resolved the dyld error and tested your patch. Unfortunately the
access-keys (ALT + key ) are still not working.


> On Tue, Aug 1, 2017 at 2:45 PM, khushboo Vashi <khushboo.vashi@gmail.com>
> wrote:
>
>>
>>
>> On Tue, Aug 1, 2017 at 2:43 PM, Konstantin Tokarev <

**Comment by sl1pkn07** [comment]
@khushboovashi how your resolve the rpath issue? with CMAKE_MACOSX_RPATH=OFF?

**Comment by annulen** [comment]
Fixed in abbe3830f1f1c912717c65f493fb71a5905268b8 and b519900c26b92e1f92edcdf33734e8d9470f4342

**Comment by khushboovashi** [comment]
Thanks for working on this.
